### ISSUE TITLE
False positive - an overloaded method is reported to be unsubscriptable

### ISSUE THREAD

**Issue by AChenQ** [PyCQA/pylint#5189]
### Bug description

```python
# pylint:disable=missing-module-docstring
# pylint:disable=missing-class-docstring
# pylint:disable=missing-function-docstring
# pylint:disable=too-few-public-methods

from typing import List, Sequence, Union, overload

_T = List[int]

class Dataset:
    data: List[_T] = [[i] for i in range(5)]

    @overload
    def __getitem__(self, index: int) -> _T:
        ...

    @overload
    def __getitem__(self, index: slice) -> Sequence[_T]:
        ...

    def __getitem__(self, index: Union[int, slice]) -> Union[_T, Sequence[_T]]:
        return self.data[index]


class_a = Dataset()
segment: _T = class_a[0]
data = segment[0] # pylint raise an unsubscriptable-object error here
```

### Pylint output

```shell
************* Module a
a.py:27:7: E1136: Value 'segment' is unsubscriptable (unsubscriptable-object)

------------------------------------------------------------------
Your code has been rated at 6.15/10 (previous run: 8.89/10, -2.74)
```

### Expected behavior

No pylint error report.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.7.9 (default, Aug 31 2020, 07:22:35)
```

### OS / Environment

Mac OS

**Comment by jacobtylerwalls** [comment]
Partner to PyCQA/astroid#1015

**Comment by jacobtylerwalls** [comment]
In fact, we are now closing `@overload` problems as duplicates of PyCQA/astroid#1015, as they share a root cause.
